ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/state/SyncState.java

```diff
@@ -311,13 +316,29 @@ public Optional<Checkpoint> getCheckpoint() {
     return checkpoint;
   }
 
+  public boolean isInitialSyncPhaseDone() {
+    return isInitialSyncPhaseDone;
+  }
+
   public void markInitialSyncPhaseAsDone() {
     isInitialSyncPhaseDone = true;
     completionListenerSubscribers.forEach(InitialSyncCompletionListener::onInitialSyncCompleted);
   }
 
-  public boolean isInitialSyncPhaseDone() {
-    return isInitialSyncPhaseDone;
+  public boolean isResyncNeeded() {
+    return isResyncNeeded;
```

Comment:
we never reset this value afaict.  once we need a resync, we always need a resync ;)

Reply:
fixed I switch this flag on `markInitialSyncPhaseAsDone`